### PR TITLE
Gpl 859 2 add support for creating uuids for non beckman samples

### LINKED ISSUE #285: GPL-859-2 - Add support for creating uuids for non-beckman samples
**User story**
Add support for creating uuids for non-beckman samples. As part of the work to improve joining between the lighthouse sample and sample table.

**Who are the primary contacts for this story**
e.g. John S (don't include surnames in public repos)

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] add a list of acceptance criteria here
- [ ] ...

**Dependencies**
This story is blocked by the following dependencies:
- #<issue_no.>
- sanger/<repo>#<issue_no.>

**Additional context**

### PR DESCRIPTION
**Pull request by stevieing**
Closes #285 

Changes proposed in this pull request:

*
*
* ...

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
 
 **NB**: Make sure to be in the virtual environment (`pipenv shell`) before running the tests.
 
+### Running tests with docker
+
+If you are unable to run tests locally because of pyodbc you can use the docker-compose
+
+        docker-compose up
```
Comment (pjvv) [inline README.md:157]: The newer docker uses:
```
docker compose up
```

File: README.md [hunk 2]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
...
+
+2. you will need to run the sqlserver with:
```
Comment (pjvv) [inline README.md:159]: This just creates the database, the SQLServer is running since it's a service in the compose file.

File: README.md [hunk 3]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
...
+
+        docker exec -ti <container_id for lighthouse> bash
```
Comment (pjvv) [inline README.md:161]:
```suggestion
        docker exec -ti <container_id for lighthouse> python ./setup_sqlserver_test_db.py
```

File: README.md [hunk 4]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
...
+
+        python ./setup_sqlserver_test_db.py
```
Comment (pjvv) [inline README.md:163]: This line is then not needed.

File: README.md [hunk 5]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
...
+
+3. you can then run the tests (with hot reloading) using:
+
+        docker exec -ti <container_id> python -m pytest -vs
```
Comment (pjvv) [inline README.md:167]: The `x` flag is handy because it stops immediately.

File: README.md [hunk 6]
```diff
@@ -150,6 +150,25 @@ A wrapper is provided with pipenv (look in the Pipfile's `[scripts]` block for m
...
+
+#TODO: I tried to run the sql server setup as part of the docker-compose using an entry point but it failed with unable to connect error.
```
Comment (pjvv) [inline README.md:169]: Yeah I don't think there is a way to wait for services to be ready without some scripts. I've written some in the past but there are some more popular ones now. e.g.: https://github.com/eficode/wait-for.

File: lighthouse/helpers/plates.py [hunk 7]
```diff
@@ -265,6 +265,7 @@ def create_post_body(barcode: str, samples: List[Dict[str, str]]) -> Dict[str, A
     phenotype = None
     description = None
     for sample in samples:
+        print(sample)
```
Comment (pjvv) [inline lighthouse/helpers/plates.py:268]: Remove this comment.

File: docker-compose.yml [hunk 8]
```diff
@@ -34,6 +34,21 @@ services:
...
+  # we need to set the localhost and root password as they can be different
+  lighthouse:
+    build: .
+    environment:
+      - LOCALHOST=host.docker.internal
+      - FLASK_APP=lighthouse
+      - FLASK_ENV=development
+      - EVE_SETTINGS=development.py
+      - ROOT_PASSWORD=
+      - SETTINGS_MODULE=lighthouse.config.development
```
Comment (pjvv) [inline docker-compose.yml:48]: I don't think this is used anywhere?

File: docker-compose.yml [hunk 9]
```diff
@@ -34,6 +34,21 @@ services:
...
+   # Setup the runner
+  # we need to set the localhost and root password as they can be different
+  lighthouse:
+    build: .
+    environment:
+      - LOCALHOST=host.docker.internal
+      - FLASK_APP=lighthouse
+      - FLASK_ENV=development
+      - EVE_SETTINGS=development.py
+      - ROOT_PASSWORD=
```
Comment (pjvv) [inline docker-compose.yml:47]: Where is this used?

**Review by pjvv — CHANGES_REQUESTED** [review]

File: README.md (hunk 5 above)
Comment (stevieing) [reply README.md:165]: I must admit I prefer it without the x.

File: docker-compose.yml (hunk 9 above)
Comment (stevieing) [reply docker-compose.yml:47]: Without it nothing works for me. It uses them as environment variables. I couldn't get it to read the .env file.

File: README.md (hunk 2 above)
Comment (pjvv) [reply README.md:159]: Did you want to update this comment?

File: README.md (hunk 5 above)
Comment (pjvv) [reply README.md:165]: Confessions are welcome. :P

File: docker-compose.yml (hunk 9 above)
Comment (pjvv) [reply docker-compose.yml:47]: But I can't find a reference in the codebase for `ROOT_PASSWORD`?

**Review by pjvv — APPROVED** [review]

**Pull request merged**